autoload/pandoc/hypertext.vim

```diff
@@ -35,7 +35,7 @@ function! pandoc#hypertext#Init() abort
     if exists('g:pandoc#hypertext#inline_link_regex')
         let s:inline_link_regex = g:pandoc#hypertext#inline_link_regex
     else
-        let s:inline_link_regex = '\v!?\[[^]]+\]\(([^) \t]+).*\)'
+        let s:inline_link_regex = '\v!?\[[^]]+\]\(([^) \t]+)\)'
```

Comment:
Any idea what the *reason* was this was coded the way it was? Is there a link syntax that includes keywords or other information inside the parenthesis after the URL? This seems like a very deliberate nested construction not an oversight. Perhaps it needs to be`[^)]*` instead of `.*` but I'm hesitant to just ditch this without knowing what it was trying to fix in the first place.

Reply:
looks like it was probably there to support [titles](https://pandoc.org/MANUAL.html#inline-links), which are a double-quoted string after the url. the current regex stops scanning the url at any whitespace, which breaks if you could have spaces in your url (pandoc itself has no issue with spaces).

have changed the regex again to try to properly parse the url separately from the label (if one exists); i've tested this with multiple links, both with urls containing spaces and double quotes, on the same line and it seems to work (as well as simpler cases). i think the only thing that would break this is escaped double quotes inside the label.

referenced-links also seem to lack space support; perhaps they could have a similar fix.

Reply:
also seems to be an issue with opening links to alternate-based files with quotes in their names, eg. `[my link](my\"file\".html)` which isn't an issue if the file is markdown, but it at least isn't an issue with the url-parsing.